try to make includegraphics insensitive to the case of the file's extension

christianp opened the pull request:
A package had a file `graphic.JPG`. The command
`\includegraphics{graphic}` didn't find it, because it was only looking for `graphic.jpg`.

This changes the `\includegraphics` command to glob all files whose names start with the given name, and then return the first file whose extension, converted to lower case, is in the expected list.

christianp committed:
try to make includegraphics insensitive to the case of the file's extension

A package had a file `graphic.JPG`. The command
`\includegraphics{graphic}` didn't find it, because it was only looking
for `graphic.jpg`.

This changes the `\includegraphics` command to glob all files whose
names start with the given name, and then return the first file whose
extension, converted to lower case, is in the expected list.

PatrickMassot commented:
I’m not sure about this change. I think your new code will always fail if the provided path includes a directory, say `\includegraphics{myfolder/myimage}`. Also if you add `''` in front of `ext` then it becomes unnecessary to add it again a few lines later.

christianp committed:
FunctionalTests.py: create directories when copying files

It looks like unit tests have until now assumed that all files are at
the top of the 'extras' directory.

christianp committed:
add a test for includegraphics with a subdirectory

christianp committed:
includegraphics: don't add '' to ext twice

The list of extensions to try needs to include the empty string in both
the graphicspath and kpsewhich blocks, so it's done at the top of the
function now.

christianp committed:
fix includegraphics when given an exact filename in a graphicspath 

The loop to check files in each of the registered graphicspath was only looking at files with a `.` character after the given filename, so if the author gave the full filename it wasn't found.

This changes the glob to look for any file with the given string as a prefix.
